Common/src/geometry/CGeometry.cpp

```diff
@@ -2456,16 +2456,78 @@ su2double CGeometry::GetSurfaceArea(const CConfig* config, unsigned short val_ma
 void CGeometry::ComputeModifiedSymmetryNormals(const CConfig* config) {
   /* Check how many symmetry planes there are and use the first (lowest ID) as the basis to orthogonalize against.
    * All nodes that are shared by multiple symmetries have to get a corrected normal. */
+
+  symmetryNormals.clear();
   symmetryNormals.resize(nMarker);
   std::vector<unsigned short> symMarkers;
 
+  /*--- Compute if marker is a straight line or plane. ---*/
+  ComputeSurfStraightness(config, false);
+
+  /*--- Check which markers are symmetry or Euler wall and store in a list. ---*/
   for (auto iMarker = 0u; iMarker < nMarker; ++iMarker) {
     if ((config->GetMarker_All_KindBC(iMarker) == SYMMETRY_PLANE) ||
         (config->GetMarker_All_KindBC(iMarker) == EULER_WALL)) {
       symMarkers.push_back(iMarker);
     }
   }
 
+  /*--- Loop over all markers and find nodes shared on curved symmetry/Euler markers. ---*/
+  for (size_t i = 1; i < symMarkers.size(); ++i) {
+    const auto iMarker = symMarkers[i];
+    for (auto iVertex = 0ul; iVertex < nVertex[iMarker]; iVertex++) {
+      const auto iPoint = vertex[iMarker][iVertex]->GetNode();
+
+      /*--- Halo points do not need to be considered. ---*/
+      if (!nodes->GetDomain(iPoint)) continue;
+
+      /*--- Get the vertex normal on the current symmetry. ---*/
+      std::array<su2double, MAXNDIM> iNormal = {};
+      std::array<su2double, MAXNDIM> kNormal = {};
+      vertex[iMarker][iVertex]->GetNormal(iNormal.data());
+
+      /*--- Loop over previous symmetries and if this point shares them, sum the normals. ---*/
+      for (size_t j = 0; j < i; ++j) {
+        const auto jMarker = symMarkers[j];
+        const auto jVertex = nodes->GetVertex(iPoint, jMarker);
+        if (jVertex < 0) continue;
+
+        /*--- If both symmetries are curved, we sum the normals, else continue the search. ---*/
+        if (boundIsStraight[iMarker] || boundIsStraight[jMarker]) continue;
+
+        std::array<su2double, MAXNDIM> jNormal = {};
+        const auto it = symmetryNormals[jMarker].find(jVertex);
+
+        if (it != symmetryNormals[jMarker].end()) {
+          jNormal = it->second;
+        } else {
+          vertex[jMarker][jVertex]->GetNormal(jNormal.data());
+        }
+
+        /*--- Average the normals on the shared nodes. ---*/
+        for (auto iDim = 0u; iDim < nDim; iDim++) kNormal[iDim] = iNormal[iDim] + jNormal[iDim];
+
+        for (auto iDim = 0u; iDim < nDim; iDim++) {
+          symmetryNormals[iMarker][iVertex][iDim] += kNormal[iDim];
+          symmetryNormals[jMarker][jVertex][iDim] += kNormal[iDim];
+        }
```

Comment:
I still don't think this is correct, I'll write it someway that expresses what we want better.

Reply:
Thanks. 
I thought what we should do is for a vertex shared by markers m0,m1,m2 we find the shared markers (m1,m0), (m2,m0) and (m2,m1) and we sum (n1+n0) + (n2+n0) + (n2+n1) = 2(n0+n1+n2)  (and then we normalize).